### PR TITLE
move extension loading to tool project, improve extension.dib loading

### PR DESCRIPTION
**Pull request by jonsequitur**
This moves the extension loading mechanism to `dotnet-interactive` from `Microsoft.DotNet.Interactive` and consolidates the `extension.dib` loading mechanism to use the same functionality used by `#!import`.

### CODE REVIEW

File: src/dotnet-interactive/CommandLine/CommandLineParser.cs [hunk 1]
```diff
@@ -504,12 +505,7 @@ private static CompositeKernel CreateKernel(string defaultKernelName, FrontendEn
         kernel.AddKernelConnector(new ConnectNamedPipeCommand());
         kernel.AddKernelConnector(new ConnectSignalRCommand());
         kernel.AddKernelConnector(new ConnectStdIoCommand(startupOptions.KernelHost));
-
```
Comment (colombod) [inline src/dotnet-interactive/CommandLine/CommandLineParser.cs:507]: why is this gone?

File: src/dotnet-interactive/PackageDirectoryExtensionLoader.cs [hunk 2]
```diff
@@ -12,11 +12,11 @@
 using Microsoft.DotNet.Interactive.Events;
 using Microsoft.DotNet.Interactive.Utility;
 using Pocket;
-using static Pocket.Logger<Microsoft.DotNet.Interactive.PackageDirectoryExtensionLoader>;
+using static Pocket.Logger<Microsoft.DotNet.Interactive.App.PackageDirectoryExtensionLoader>;
 
-namespace Microsoft.DotNet.Interactive;
+namespace Microsoft.DotNet.Interactive.App;
 
-internal class PackageDirectoryExtensionLoader : IKernelExtensionLoader
```
Comment (colombod) [inline src/dotnet-interactive/PackageDirectoryExtensionLoader.cs:19]: how is this portable to .net repl and other custom app hosting .NET Interactive?

Comment (jonsequitur) [reply src/dotnet-interactive/PackageDirectoryExtensionLoader.cs:19]: Custom apps should implement their own extensibility mechanisms.

I think factoring this out into a separate small package is worthwhile so .NET REPL can continue to use it.

Comment (colombod) [reply src/dotnet-interactive/PackageDirectoryExtensionLoader.cs:19]: Yes, totally in favour of a standalone package that could be opted in or implemented custom.

File: src/dotnet-interactive/CommandLine/CommandLineParser.cs (hunk 1 above)
Comment (jonsequitur) [reply src/dotnet-interactive/CommandLine/CommandLineParser.cs:507]: This wasn't really doing anything useful outside tests. I moved these older diagnostic methods over to the tests and want to make some more general and internally consistent diagnostic improvements.

**Review by colombod — APPROVED** [review]

**Pull request merged**